k4EDM4hep2LcioConv/src/k4Lcio2EDM4hepConv.cpp

```diff
@@ -589,11 +614,13 @@ namespace LCIO2EDM4hepConv {
       }
     }
     // Filling all the OneToMany and OnToOne Relations and creating the AssociationCollections.
+    auto calocontr = createCaloHitContributions(typeMapping.simCaloHits, typeMapping.mcParticles);
```

Comment:
Move this further down, so that the creation of this collection and putting it into the event are closer together.